gammapy/datasets/simulate.py

```diff
@@ -316,13 +320,13 @@ def run(self, dataset, observation=None):
         if len(dataset.models) > 1:
             events_src = self.sample_sources(dataset)
 
-            if dataset.psf:
+            if dataset.psf and (len(events_src.table) > 0):
```

Comment:
Is the condition `(len(events_src.table) > 0)` useful here or should it apply to both:
```
if len(events_src.table) > 0:
    if dataset.edisp:
...
    else:
...
    if dataset.psf:
...
```

Reply:
Yes, it can be applied to both. I modified it.